Src/FluentAssertions/Xml/XmlReaderValidator.cs

```diff
@@ -103,8 +104,41 @@ private ValidationResult Validate()
                     return validationResult;
                 }
 
-                subjectReader.Read();
-                otherReader.Read();
+                var readOnSubjectReader = true;
```

Comment:
I don't really get how this works together with the `switch` statement. Shouldn't this be part of a `case`?

Reply:
You're right, this checks should be done only while `XmlNodeType.Element` is current.